src/snowflake/cli/api/sql_execution.py

```diff
@@ -42,9 +42,8 @@
 from snowflake.connector.errors import ProgrammingError
 
 
-class SqlExecutor:
-    def __init__(self, connection: SnowflakeConnection | None = None):
-        self._snowpark_session = None
+class BaseSqlExecutor:
```

Comment:
How about we take the opportunity to simply the various "execute" variants in here? I'm not sure why we need all these private helpers. From the POV of mocking the contract, it would be a lot easier if we just had a single point to mock (I know that everything eventually rolls up to _execute_string, but there's a rather complex set of dependencies between the calls).